exercicios/para-casa/entregas/Sabrina-Lopes/resolucao.js

```diff
@@ -0,0 +1,72 @@
+export class SuperPoder {
+  #nome;
+  #categoria;
+
+  constructor(nome, categoria) {
+    if (typeof categoria !== 'number') {
+      throw new Error("A categoria deve ser um número");
+    }
+
+    this.#nome = nome;
+    this.#categoria = categoria;
+  }
+
+  get nome() {
+    return this.#nome;
+  }
+
+  get categoria() {
+    return this.#categoria;
+  }
+}
+
+export class Personagem {
+  #nome;
+  #nomeVidaReal;
+  #poderes;
+
+  constructor(nome, nomeVidaReal) {
+    this.#nome = nome;
+    this.#nomeVidaReal = nomeVidaReal;
+    this.#poderes = [];
+  }
+
+  get nome() {
+    return this.#nome;
+  }
+
+  get nomeVidaReal() {
+    return this.#nomeVidaReal;
+  }
+
+  adicionarSuperPoder(superPoder) {
+    if (this.#poderes.length < 4) {
+      this.#poderes.push(superPoder);
+    } else {
+      console.log(`O ${this.#nome} já tem o número máximo de poderes (4).`);
+    }
+  }
+
+  poderTotal() {
+    let total = 0;
+    for (const superPoder of this.#poderes) {
+      total += superPoder.categoria;
+    }
+    return total;
+  }
```

Comment:
que interessante o uso do for of aqui, jamais teria pensado nisso... muito legal de ver as várias formas de resolução do mesmo exercício :)